### PR TITLE
Dilpreet | Test | Stock Transaction Report

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/meta/success.html [hunk 1]
```diff
@@ -30,7 +30,7 @@ <h5>You can close this window.</h5>
 
     <div class="copy-right">
         <p>
-            All right are reserved &copy; 2020 Walkover <a href="https://app.giddh.com/" target="_Blank">Giddh.com</a>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/meta/success.html]: revert

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 2]
```diff
@@ -743,7 +743,7 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
                 if (isElectron) {
                     this.router.navigate(['/login']);
                 } else {
-                    window.location.href = (environment.production) ? `https://giddh.com/login/?action=logout` : `https://test.giddh.com/login/?action=logout`;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/header.component.ts]: revert

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.html [hunk 3]
```diff
@@ -137,61 +185,152 @@
...
                             <h5>
                                 Inward
                                 <div class="mt-2">
-                                    <span class="d-inline-flex number-total"> 432.00 </span>
+                                    <span
+                                        class="d-inline-flex number-total w-100"
+                                        [matTooltip]="stockTransactionReportBalance?.inwards?.amount"
+                                    >
+                                        {{
+                                            stockTransactionReportBalance?.inwards?.amount
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.html:285]: use amount-field component

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.html [hunk 4]
```diff
@@ -137,61 +185,152 @@
...
                             </h5>
                         </div>
                         <div class="total-details-box mr-r05 border-none payment-receipt">
                             <h5>
                                 Inward
                                 <div class="mt-2">
-                                    <span class="d-inline-flex number-total"> 432.00 </span>
+                                    <span
+                                        class="d-inline-flex number-total w-100"
+                                        [matTooltip]="stockTransactionReportBalance?.inwards?.amount"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.html]: prepend currency code in tooltip

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.html [hunk 5]
```diff
@@ -202,87 +341,191 @@ <h5>
...
+                        </th>
                         <td data-title="Account Name" class="text-right" mat-cell *matCellDef="let element">
-                            {{ element.accountName }}
+                            {{ element?.account?.name ?? "-" }}
+                        </td>
+                    </ng-container>
+                    <ng-container matColumnDef="stockName">
+                        <th class="text-right" mat-header-cell *matHeaderCellDef mat-sort-header>Stock Name</th>
+                        <td data-title="Stock Name" class="text-right" mat-cell *matCellDef="let element">
+                            {{ element?.stockNameUniqueName?.name ?? "-" }}
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.html]:
```suggestion
                            {{ element?.stock?.name ?? "-" }}
```

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.html [hunk 6]
```diff
@@ -202,87 +341,191 @@ <h5>
...
-                            {{ element?.value }}
+                        <th class="text-right" mat-header-cell *matHeaderCellDef mat-sort-header>Value</th>
+                        <td
+                            data-title="Value"
+                            class="text-right"
+                            mat-cell
+                            *matCellDef="let element"
+                            [matTooltip]="element?.amount"
+                        >
+                            {{ element?.amount ? (element?.amount | giddhCurrency) : "-" }}
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.html]: use amount field component for amount everywhere

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 7]
```diff
@@ -7,104 +7,60 @@ import { GIDDH_DATE_FORMAT, GIDDH_NEW_DATE_FORMAT_UI } from "../../../shared/hel
...
+    /** Instance of  account name searching column */
+    @ViewChild('accountName', { static: true }) public accountName: ElementRef;
+    /** Instance of sort header */
+    @ViewChild(MatSort) sort: MatSort;
+    /** This will use for instance of warehouses Dropdown */
+    public warehousesDropdown: FormControl = new FormControl();
+    /** This will use for instance of branches Dropdown */
+    public branchesDropdown: FormControl = new FormControl();
+    /** Thsi will use for searching for stock */
+    public accountNameSearching: FormControl = new FormControl();
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts]: rename

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 8]
```diff
@@ -7,104 +7,60 @@ import { GIDDH_DATE_FORMAT, GIDDH_NEW_DATE_FORMAT_UI } from "../../../shared/hel
...
+    /** Search field form control */
+    public searchingFilters: FormControl = new FormControl();
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts]: rename

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 9]
```diff
@@ -117,50 +73,378 @@ export class InventoryTransactionListComponent implements OnInit {
...
+     */
+    public selectChiplistValue(option: any): void {
+        const selectOptionValue = option?.option?.value;
+        this.allowAddChip = false;
+        setTimeout(() => {
+            this.allowAddChip = true;
+        }, 300);
+        if (option?.option?.value?.type === 'STOCK GROUP') {
+            if (this.stockReportRequest.stockGroupUniqueNames?.length > 0) {
+                this.stockReportRequest.stockGroupUniqueNames = option?.option?.value?.uniqueName;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts]: change

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 10]
```diff
@@ -117,50 +73,378 @@ export class InventoryTransactionListComponent implements OnInit {
...
+                return;
+            } else {
+                this.stockReportRequest.stockGroupUniqueNames?.push(option?.option?.value?.uniqueName);
+            }
+        } else if (option?.option?.value?.type === 'STOCK') {
+            const findStockColumnCheck = this.customiseColumns?.find(value => value?.value === "stockName");
+            if (this.stockReportRequest.stockUniqueNames?.length == 0 && findStockColumnCheck?.checked) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts]:
```suggestion
            if (this.stockReportRequest.stockUniqueNames?.length === 0 && findStockColumnCheck?.checked) {
```

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 11]
```diff
@@ -117,50 +73,378 @@ export class InventoryTransactionListComponent implements OnInit {
...
+                findStockColumnCheck.checked = false;
+                this.displayedColumns = this.displayedColumns?.filter(value => value !== "stockName");
+            } else if (this.stockReportRequest.stockUniqueNames?.length > 0 && !findStockColumnCheck?.checked) {
+                findStockColumnCheck.checked = true;
+                this.filteredDisplayColumns();
+            }
+            this.stockReportRequest.stockUniqueNames?.push(option?.option?.value?.uniqueName);
+        } else {
+            const findVariantColumnCheck = this.customiseColumns?.find(value => value?.value === "variantName");
+            if (this.stockReportRequest.variantUniqueNames?.length == 0 && findVariantColumnCheck?.checked) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts]:
```suggestion
            if (this.stockReportRequest.variantUniqueNames?.length === 0 && findVariantColumnCheck?.checked) {
```

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 12]
```diff
@@ -117,50 +73,378 @@ export class InventoryTransactionListComponent implements OnInit {
...
+            this.changeDetection.detectChanges();
+        }
+    }
+
+    /**
+     * This will use for get stock transaction report filters
+     *
+     * @memberof InventoryTransactionListComponent
+     */
+    public getStockTransactionReportFilters(): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts]: rename

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 13]
```diff
@@ -117,50 +73,378 @@ export class InventoryTransactionListComponent implements OnInit {
...
+        this.searchStockReportRequest.variantUniqueNames = this.stockReportRequest.variantUniqueNames;
+        this.inventoryService.searchStockTransactionReport(this.searchStockReportRequest).pipe(take(1)).subscribe(response => {
+            if (response && response.body && response.status === 'success') {
+                this.fieldFilteredOptions = response.body.results;
+                this.searchStockReportRequest.page = response.body.page;
+                this.searchStockReportRequest.count = response.body.count;
+                this.searchStockReportRequest.count = response.body.totalItems;
+                this.searchStockReportRequest.count = response.body.totalPages;
+            } else {
+                this.toaster.showSnackBar("success", response?.body);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts]:
```suggestion
                this.toaster.showSnackBar("warning", response?.body);
```

Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts]: reset report response here

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 14]
```diff
@@ -117,50 +73,378 @@ export class InventoryTransactionListComponent implements OnInit {
...
+    public getStockTransactionReportFilters(): void {
+        this.searchStockReportRequest.stockGroupUniqueNames = this.stockReportRequest.stockGroupUniqueNames;
+        this.searchStockReportRequest.stockUniqueNames = this.stockReportRequest.stockUniqueNames;
+        this.searchStockReportRequest.variantUniqueNames = this.stockReportRequest.variantUniqueNames;
+        this.inventoryService.searchStockTransactionReport(this.searchStockReportRequest).pipe(take(1)).subscribe(response => {
+            if (response && response.body && response.status === 'success') {
+                this.fieldFilteredOptions = response.body.results;
+                this.searchStockReportRequest.page = response.body.page;
+                this.searchStockReportRequest.count = response.body.count;
+                this.searchStockReportRequest.count = response.body.totalItems;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts]:
```suggestion
                this.searchStockReportRequest.totalItems = response.body.totalItems;
```

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 15]
```diff
@@ -117,50 +73,378 @@ export class InventoryTransactionListComponent implements OnInit {
...
+                this.searchStockReportRequest.count = response.body.totalPages;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts]:
```suggestion
                this.searchStockReportRequest.totalPages = response.body.totalPages;
```

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 16]
```diff
@@ -117,50 +73,378 @@ export class InventoryTransactionListComponent implements OnInit {
...
+            this.isLoading = false;
+            if (response && response.body && response.status === 'success') {
+                this.dataSource = response.body.transactions;
+                this.stockReportRequest.page = response.body.page;
+                this.stockReportRequest.totalItems = response.body.totalItems;
+                this.stockReportRequest.totalPages = response.body.totalPages;
+                this.stockReportRequest.count = response.body.count;
+            } else {
+                this.dataSource = [];
+                this.stockReportRequest.totalItems = 0;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts:437]: reset totalPages also

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts [hunk 17]
```diff
@@ -117,50 +73,378 @@ export class InventoryTransactionListComponent implements OnInit {
...
+        delete this.stockReportRequest.count;
+        delete this.stockReportRequest.page;
+        delete this.stockReportRequest.transactionType;
+        delete this.stockReportRequest.totalItems;
+        delete this.stockReportRequest.totalPages;
+        this.inventoryService.getStockTransactionReportBalance(cloneDeep(this.stockReportRequest)).pipe(take(1)).subscribe(response => {
+            if (response && response.body && response.status === 'success') {
+                this.stockTransactionReportBalance = response.body;
+            } else {
+                this.toaster.showSnackBar("error", response?.message);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts:452]: reset the balance

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts (hunk 16 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/new-inventory/component/inventory-transaction-list/inventory-transaction-list.component.ts:437]: no using this

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**